### PR TITLE
Fix subresource removing from the resource list

### PR DESCRIPTION
**Pull request by anttiylitokola**

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@yogpan01, @TeroJaasko

**Comment by yogpan01** [comment]
LGTM

File: source/libNsdl/src/sn_grs.c [hunk 1]
```diff
@@ -764,11 +763,8 @@ sn_nsdl_resource_info_s *sn_grs_search_resource(struct grs_s *handle, uint16_t p
     else if (search_method == SN_GRS_DELETE_METHOD) {
         /* Scan all nodes on list */
         ns_list_foreach(sn_nsdl_resource_info_s, resource_search_temp, &handle->resource_root_list) {
-            uint8_t *temp_ptr = resource_search_temp->path;
-
-            /* If found, return pointer */
-            if ((*(temp_ptr + (uint8_t)pathlen) == '/')
-                    && !memcmp(resource_search_temp->path, path_temp_ptr, pathlen)) {
+            if (resource_search_temp->pathlen > pathlen &&
+                    0 == memcmp(resource_search_temp->path, path, pathlen)) {
```
Comment (TeroJaasko) [inline source/libNsdl/src/sn_grs.c]: Doesn't this return also resources with same prefix, eg. "dr/x" would find also "dr/xyz/1"?

Comment (anttiylitokola) [inline source/libNsdl/src/sn_grs.c]: You are right about that. Need to fix that part.

**Comment by anttiylitokola** [comment]
rebuild

**Pull request merged**